src/main/java/net/revelc/code/impsort/maven/plugin/AbstractImpSortMojo.java

```diff
@@ -239,14 +279,27 @@ public final void execute() throws MojoExecutionException, MojoFailureException
         getLog().debug("Reading file " + path);
 
         try {
-          Result result = impSort.parseFile(path);
-          result.getImports().forEach(imp -> getLog().debug("Found import: " + imp));
-          if (result.isSorted()) {
+          byte[] buf = Files.readAllBytes(path);
+          String newHash = getHash(buf);
+          String key = baseDir.relativize(path).toString();
```

Comment:
This relativize will only make the cache key relative to the root of each module in a multi-module project. If the cache is reused between modules (such as when an absolute path is provided for the cachedir) and files have the same name in different modules, then this may result in cache entries being overwritten and files reformatted, even though they haven't changed. Would it be better to use the absolute paths, the path relative to the root of a multi-module project, or keep it relative to each module and accept that same-name files in different modules won't benefit from this cache?

Reply:
Fwiw, I think the same problem applies to the `formatter` maven plugin too (maybe even worse).
The safest way would be to use the canonical path to store the data.

Reply:
Yeah, I was thinking the same. I'm not familiar with formatter-maven-plugin's implementation. I'd be in favor of switching to use the canonical path here. The one downside is that the cache couldn't be relocated to be reused on a different machine or git clone... but I think that's okay, since this is really an optimization to help speed up in-place development.